### PR TITLE
Initial commit of sync Kotlin sample.

### PR DESCRIPTION
**Pull request by garanj**
Initial commit of a sample to demonstrate synchronizing data to Google Fit using WorkManager, for both ad-hoc and periodic syncs.

### CODE REVIEW

File: SyncKotlin/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+The sample also demonstrates:
+
+-   Allowing for user-initiated immediate syncs
+-   Handling transient errors with the upload, and retrying
+-   Recording failure and using [notifications](https://developer.android.com/guide/topics/ui/notifiers/notifications)
+    to alert the user.
+-   Turning off the synchronisation after retries are exhausted and the
+    sync process ultimately fails.
+
+In this sample, a toy database `StepsDatabase` is used as the source of data to
```
Comment (codingjeremy) [inline SyncKotlin/README.md:32]: Change 'toy' database with 'demo', 'local', or remove. Toy is a bit confusing to me, as I'm thinking about kid toys. :)

File: SyncKotlin/app/src/main/res/drawable-v24/ic_launcher_foreground.xml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (codingjeremy) [inline SyncKotlin/app/src/main/res/drawable-v24/ic_launcher_foreground.xml:1]: Missing open source header.

File: SyncKotlin/app/src/main/res/drawable/ic_launcher_background.xml [hunk 3]
```diff
@@ -0,0 +1,170 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (codingjeremy) [inline SyncKotlin/app/src/main/res/drawable/ic_launcher_background.xml:1]: Missing open source header.

File: SyncKotlin/app/src/main/res/drawable/small_icon.xml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (codingjeremy) [inline SyncKotlin/app/src/main/res/drawable/small_icon.xml:1]: Missing open source header.

File: SyncKotlin/app/src/main/res/mipmap-anydpi-v26/ic_launcher.xml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (codingjeremy) [inline SyncKotlin/app/src/main/res/mipmap-anydpi-v26/ic_launcher.xml:1]: Missing open source header.

File: SyncKotlin/app/src/main/res/mipmap-anydpi-v26/ic_launcher_round.xml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (codingjeremy) [inline SyncKotlin/app/src/main/res/mipmap-anydpi-v26/ic_launcher_round.xml:1]: Missing open source header.

File: SyncKotlin/app/src/main/java/com/google/android/gms/fit/samples/synckotlin/MainActivity.kt [hunk 7]
```diff
@@ -0,0 +1,335 @@
...
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.material.snackbar.Snackbar
+
+enum class FitActionRequestCode {
+    ENABLE_REPEATED_SYNC,
+    IMMEDIATE_SYNC
+}
+
+const val TAG = "FitSync"
+
```
Comment (codingjeremy) [inline SyncKotlin/app/src/main/java/com/google/android/gms/fit/samples/synckotlin/MainActivity.kt:52]: All classes should have a short explanation/overview at the start of class.

File: SyncKotlin/app/src/main/java/com/google/android/gms/fit/samples/synckotlin/MainActivity.kt [hunk 8]
```diff
@@ -0,0 +1,335 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        sharedPreferences = getSharedPreferences(
+            getString(R.string.preference_file_key),
+            Context.MODE_PRIVATE
+        )
+
+        syncToggle = findViewById(R.id.fitSyncSwitch)
```
Comment (codingjeremy) [inline SyncKotlin/app/src/main/java/com/google/android/gms/fit/samples/synckotlin/MainActivity.kt:69]: Optional: Might want to move to databinding or viewbinding but that can be for another PR.

File: SyncKotlin/app/src/main/java/com/google/android/gms/fit/samples/synckotlin/Notifications.kt [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import kotlin.random.Random
+
+const val CHANNEL_ID = "FitSync"
+
```
Comment (codingjeremy) [inline SyncKotlin/app/src/main/java/com/google/android/gms/fit/samples/synckotlin/Notifications.kt:30]: Our internal style guidelines require a comment above the new class that gives a short overview of what the class does.

File: SyncKotlin/app/src/main/java/com/google/android/gms/fit/samples/synckotlin/Permissions.kt [hunk 10]
```diff
@@ -0,0 +1,50 @@
+/*
```
Comment (codingjeremy) [inline SyncKotlin/app/src/main/java/com/google/android/gms/fit/samples/synckotlin/Permissions.kt:1]: I would rename this file to Utils.kt or PermissionUtils.kt or something.

File: SyncKotlin/app/src/main/java/com/google/android/gms/fit/samples/synckotlin/FitSync.kt [hunk 11]
```diff
@@ -0,0 +1,227 @@
...
+        }
+    }
+
+    private fun performSync(account: GoogleSignInAccount) {
+        Log.i(TAG, "Performing sync")
+        // Get the last sync time
+        val lastSyncMillis =
+            sharedPreferences.getLong(appContext.getString(R.string.last_sync_timestamp), 0)
+
+        // Get any steps since lastSyncMillis. StepsDatabase is a class just for the purposes of
```
Comment (codingjeremy) [inline SyncKotlin/app/src/main/java/com/google/android/gms/fit/samples/synckotlin/FitSync.kt:131]: I would call out it's a mock database, so it's clear right away that this isn't an implementation at all.

File: SyncKotlin/app/src/main/java/com/google/android/gms/fit/samples/synckotlin/StepsDatabase.kt [hunk 12]
```diff
@@ -0,0 +1,32 @@
+/*
```
Comment (codingjeremy) [inline SyncKotlin/app/src/main/java/com/google/android/gms/fit/samples/synckotlin/StepsDatabase.kt:1]: I would rename this class to MockStepsDatabase.kt, so it's clear it isn't real maybe?

File: SyncKotlin/app/src/main/java/com/google/android/gms/fit/samples/synckotlin/FitSync.kt [hunk 13]
```diff
@@ -0,0 +1,227 @@
...
+        // this sample, to serve as a placeholder for the storage an app might use, be it local or
+        // in a app-specific cloud.
+        val db = StepsDatabase()
```
Comment (codingjeremy) [inline SyncKotlin/app/src/main/java/com/google/android/gms/fit/samples/synckotlin/FitSync.kt:134]: Optional: I usually prefer to always use full names, so it's clear to all devs, e.g., 'database'... some might not know what that stands for.

File: SyncKotlin/README.md (hunk 1 above)
Comment (codingjeremy) [reply SyncKotlin/README.md:32]: Realized later this is a mock database. I would use 'mock' instead of toy...

In this sample, a mock/fake database `MockStepsDatabase` is used as the source of data to be uploaded to Fit, which serves as a placeholder for a real local or remote repository.

File: SyncKotlin/app/src/main/java/com/google/android/gms/fit/samples/synckotlin/MainActivity.kt (hunk 8 above)
Comment (garanj) [reply SyncKotlin/app/src/main/java/com/google/android/gms/fit/samples/synckotlin/MainActivity.kt:69]: Done.

**Comment by garanj** [comment]
Apologies Jeremy! Just got back to this!!!!!! I've addressed your comments I hope!

**Review by codingjeremy — APPROVED** [review]
